perimeter/public_access.sp

```diff
@@ -461,102 +461,36 @@ control "s3_bucket_acl_prohibit_public_write_access" {
-        and s -> 'Condition' -> 'StringEquals' -> 'aws:principalorgid' is null
-        and s -> 'Condition' -> 'StringEqualsIgnoreCase' -> 'aws:principalorgid' is null
-        and (
-          s -> 'Condition' -> 'StringLike' -> 'aws:principalorgid' is null
-          or s -> 'Condition' -> 'StringLike' -> 'aws:principalorgid' ? '*'
-        )
-        -- aws:PrincipalAccount
-        and s -> 'Condition' -> 'StringEquals' -> 'aws:principalaccount' is null
-        and s -> 'Condition' -> 'StringEqualsIgnoreCase' -> 'aws:principalaccount' is null
-        and (
-          s -> 'Condition' -> 'StringLike' -> 'aws:principalaccount' is null
-          or s -> 'Condition' -> 'StringLike' -> 'aws:principalaccount' ? '*'
-        )
-        -- aws:PrincipalArn
-        and s -> 'Condition' -> 'StringEquals' -> 'aws:principalarn' is null
-        and s -> 'Condition' -> 'StringEqualsIgnoreCase' -> 'aws:principalarn' is null
-        and (
-          s -> 'Condition' -> 'StringLike' -> 'aws:principalarn' is null
-          or s -> 'Condition' -> 'StringLike' -> 'aws:principalarn' ? '*'
-        )
-        and (
-          s -> 'Condition' -> 'ArnEquals' -> 'aws:principalarn' is null
-          or s -> 'Condition' -> 'ArnEquals' -> 'aws:principalarn' ? '*'
-        )
-        and (
-          s -> 'Condition' -> 'ArnLike' -> 'aws:principalarn' is null
-          or s -> 'Condition' -> 'ArnLike' -> 'aws:principalarn' ? '*'
-        )
-        -- aws:SourceArn
-        and s -> 'Condition' -> 'StringEquals' -> 'aws:sourcearn' is null
-        and s -> 'Condition' -> 'StringEqualsIgnoreCase' -> 'aws:sourcearn' is null
-        and (
-          s -> 'Condition' -> 'StringLike' -> 'aws:sourcearn' is null
-          or s -> 'Condition' -> 'StringLike' -> 'aws:sourcearn' ? '*'
-        )
-        and (
-          s -> 'Condition' -> 'ArnEquals' -> 'aws:sourcearn' is null
-          or s -> 'Condition' -> 'ArnEquals' -> 'aws:sourcearn' ? '*'
-        )
-        and (
-          s -> 'Condition' -> 'ArnLike' -> 'aws:sourcearn' is null
-          or s -> 'Condition' -> 'ArnLike' -> 'aws:sourcearn' ? '*'
-        )
-        and (
-          s -> 'Principal' -> 'AWS' = '["*"]'
-          or s ->> 'Principal' = '*'
-        )
-      group by
-        __ARN_COLUMN__
-    )
+  resource_policy_public_sql = <<-EOT
     select
       r.__ARN_COLUMN__ as resource,
       case
-        when p.__ARN_COLUMN__ is null then 'ok'
-        else 'alarm'
+        when pa.is_public = true then 'alarm'
+        else 'ok'
       end as status,
       case
-        when p.__ARN_COLUMN__ is null then title || ' policy does not allow public access.'
-        else title || ' policy contains ' || coalesce(p.statements_num, 0) ||
-        ' statement(s) that allow public access.'
+        when pa.is_public = false then title || ' policy does not allow public access.'
+        else title || ' policy contains ' || count(pa.public_statement_ids) || ' statement(s) that allow public access.'
```

Comment:
Is there any reasonable way to show one or more statement IDs in the `reason`? For instance, could we show the first or first two statement IDs and then include something like `... and 2 more statement(s) that allow public access`?

So the alarm reason would look like something like, `Bucketname policy allows public access in my-sid-1, Statement[2], and 3 more statement(s).`